sqlmodel/main.py

```diff
@@ -166,6 +170,18 @@ def Field(
     schema_extra: Optional[Dict[str, Any]] = None,
 ) -> Any:
     current_schema_extra = schema_extra or {}
+    if default is PydanticUndefined:
+        if type(sa_column) is types.FunctionType: #lambda 
```

Comment:
There seems to be an error with this line:

> Non-overlapping identity check (left operand type: "Union[Type[Column[Any]], Type[PydanticUndefinedType]]", right operand type: "Type[FunctionType]")

Reply:
sa_column maybe a FunctionType which return a Column: https://github.com/sqlalchemy/sqlalchemy/issues/5669#issuecomment-715338925 
I'll change it.